src/Storages/IStorage.h

```diff
@@ -178,11 +178,21 @@ class IStorage : public std::enable_shared_from_this<IStorage>, public TypePromo
     using ColumnSizeByName = std::unordered_map<std::string, ColumnSize>;
     virtual ColumnSizeByName getColumnSizes() const { return {}; }
 
+    /// Get the total length of all indexes in the table.
```

Comment:
No need to change anything in C++, all existing data can be gathered from fields

```
primary_key_size
marks_bytes
secondary_indices_compressed_bytes
secondary_indices_uncompressed_bytes
secondary_indices_marks_bytes
```

in `system.parts`.